Kianoosh76/IBS
# [IBS-69] Test & migrate groups (student)

## Description
- Jira: https://ibs-utm.atlassian.net/browse/IBS-69
- Refactored Group endpoints for Students.

## Changes Made
- Refactored following endpoints`
  - `module/group/student/accept.js`
  - `module/group/student/check.js`
  - `module/group/student/create.js`
  - `module/group/student/disinvite.js`
  - `module/group/student/invite.js`
  - `module/group/student/leave.js`
  - `module/group/student/reject.js`
- Enabled the disinvite endpoint for a student in `backend/route/student.js`for refactoring purposes

## Test Guidelines
- In `/backend/` run the following and ensure that they all passes:
 ```bash 
npm test group_student_accept
```
 ```bash 
npm test group_student_check
```
 ```bash 
npm test group_student_create
```
 ```bash 
npm test group_student_disinvite
```
 ```bash 
npm test group_student_invite
```
 ```bash 
npm test group_student_leave
```
 ```bash 
npm test group_student_reject
```

## Checklist

### For PR author:
- [ ] Reviewed code changes.
- [ ] Added appropriate comments.
- [ ] Ensured consistent code style and formatting.
- [ ] Included unit/integration tests.
- [ ] Handled edge cases appropriately.
- [ ] Celebrated accomplishments 🎉

### For PR reviewer:
- [ ] Reviewed code changes.
- [ ] Verified clear and helpful comments.
- [ ] Ensured code style aligns with standards.
- [ ] Examined test coverage.
- [ ] Provided feedback on potential edge cases.
